### PR TITLE
feat: duplicate chart endpoint

### PR DESCRIPTION
**Pull request by rephus**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1717

### Description:
Take optional parameter `duplicateFrom` on the save endpoint to save new chart using existing chartUuid

### Preview:
```
curl 'http://localhost:3000/api/v1/projects/3675b69e-8324-4110-bdca-059031aa8da3/saved?duplicateFrom=b14c8165-3874-4076-8258-e599398e9d35' -H 'Content-Type: application/json' 
```

```
{"status":"ok","results":{"uuid":"8b2e39b3-dbe8-4059-aa7f-1eae1459a12a","projectUuid":"3675b69e-8324-4110-bdca-059031aa8da3","name":"test","tableName":"customers","updatedAt":"2022-04-08T14:49:00.042Z","metricQuery":{"dimensions":["customers_customer_id"],"metrics":["customers_unique_customer_count"],"filters":{},"sorts":[{"fieldId":"customers_unique_customer_count","descending":true}],"limit":500,"tableCalculations":[]},"chartConfig":{"type":"cartesian","config":{"layout":{"xField":"customers_customer_id","yField":["customers_unique_customer_count"]},"eChartsConfig":{"series":[{"type":"bar","encode":{"xRef":{"field":"customers_customer_id"},"yRef":{"field":"customers_unique_customer_count"}}}]}}},"tableConfig":{"columnOrder":["customers_unique_customer_count","customers_customer_id"]}}}
```
### Scope of the changes (select all that apply):
- [ ] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

File: packages/backend/src/services/SavedChartsService/SavedChartService.ts [hunk 1]
```diff
@@ -148,4 +148,30 @@ export class SavedChartService {
...
+        const chart = await this.savedChartModel.get(chartUuid);
+        const duplicatedChart = {
+            ...chart,
+            name: `Copy of ${chart.name}`,
+        };
+        const newSavedChart = await this.savedChartModel.create(
+            projectUuid,
+            duplicatedChart,
+        );
+        analytics.track({
```
Comment (owlas) [inline packages/backend/src/services/SavedChartsService/SavedChartService.ts:169]: @TuringLovesDeathMetal is this better as another create event but with a property?

```
            event: 'saved_chart.created',
             userId: user.userUuid,
             properties: {
                 duplicated: true,
                 ...SavedChartService.getCreateEventProperties(newSavedChart)
             },
               
         });
 ```

**Review by owlas — COMMENTED** [review]
one question on analytics

File: packages/backend/src/services/SavedChartsService/SavedChartService.ts (hunk 1 above)
Comment (owlas) [reply packages/backend/src/services/SavedChartsService/SavedChartService.ts:169]: Reason if you wanted to sum chart creation events  you probably want to include duplications

Comment (TuringLovesDeathMetal) [reply packages/backend/src/services/SavedChartsService/SavedChartService.ts:169]: yes please! 

Is there some way to have two events:

- one event for `duplicated_chart_created`
- one event for `saved_chart_created`

Comment (TuringLovesDeathMetal) [reply packages/backend/src/services/SavedChartsService/SavedChartService.ts:169]: Could we also please add `saved_query_id` to the `duplicated_chart_created`

ideally something like:
`duplicate_of_saved_query_id` for the chart_id it's being created from and `new_saved_query_id` for the new chart being created

**Review by owlas — APPROVED** [review]

**Pull request merged**